LV_analysis/nbconverted/1_get_eADAGE_LV_coverage.py

```diff
@@ -0,0 +1,389 @@
+# 
+# The eADAGE model uses a DAE to extracts patterns of gene expression activity in the latent variables (referred to as nodes in the paper). Here we are examining the coverage of generic genes within these latent variables.
+# 
+# **Definitions:**
+# * Generic genes: Are genes that are consistently differentially expressed across multiple simulated experiments.
+# 
+# * Other genes: These are all other non-generic genes. These genes include those that are not consistently differentially expressed across simulated experiments - i.e. the genes are specifically changed in an experiment. It could also indicate genes that are consistently unchanged (i.e. housekeeping genes)
+
+# In[1]:
+
+
+get_ipython().run_line_magic('load_ext', 'autoreload')
+get_ipython().run_line_magic('autoreload', '2')
+
+import os
+import random
+import textwrap
+import scipy
+import pandas as pd
+import seaborn as sns
+import matplotlib.pyplot as plt
+from sklearn.preprocessing import MinMaxScaler
+
+import rpy2.robjects as ro
+from rpy2.robjects import pandas2ri
+from rpy2.robjects.conversion import localconverter
+
+from ponyo import utils
+from generic_expression_patterns_modules import lv
+
+
+# In[2]:
+
+
+# Get data directory containing gene summary data
+base_dir = os.path.abspath(os.path.join(os.getcwd(), "../"))
+data_dir = os.path.join(base_dir, "pseudomonas_analysis")
+
+# Read in config variables
+config_filename = os.path.abspath(
+    os.path.join(base_dir, "configs", "config_pseudomonas_33245.tsv")
+)
+
+params = utils.read_config(config_filename)
+
+local_dir = params["local_dir"]
+
+project_id = params["project_id"]
+
+
+# In[3]:
+
+
+# Output file
+nonzero_figure_filename = "nonzero_LV_coverage_pa.svg"
+highweight_figure_filename = "highweight_LV_coverage_pa.svg"
+
+
+# ## Load data
+
+# In[4]:
+
+
+# Get gene summary file
+summary_data_filename = os.path.join(
+    data_dir,
+    f"generic_gene_summary_{project_id}_crcZ_v_WT.tsv"
+)
+
+
+# In[5]:
+
+
+# Load gene summary data
+data = pd.read_csv(summary_data_filename, sep="\t", index_col=0, header=0)
+
+# Check that genes are unique since we will be using them as dictionary keys below
+assert(data.shape[0] == len(data["Gene ID"].unique()))
+
+
+# **Manual steps to process eADAGE data**
+# 1. Data downloaded from https://zenodo.org/record/580093
+# 2. Get Weight matrix (`eADAGE weight matrix.xlsx`) file
+# 3. Save tab (`weight matrix`) as .csv file in `local_dir`
+
+# In[6]:
+
+
+# Load eADAGE weight matrix
+eADAGE_weight_filename = os.path.join(local_dir, "eADAGE weight matrix.csv")
+eADAGE_weight = pd.read_csv(eADAGE_weight_filename, sep=",", index_col=0, header=0)
+
+
+# In[7]:
+
+
+eADAGE_weight.shape
+
+
+# In[8]:
+
+
+# Get a rough sense for how many genes contribute to a given LV
+# (i.e. how many genes have a value > 0 for a given LV)
+# Notice that eADAGE is NOT sparse
+(eADAGE_weight != 0).sum().sort_values(ascending=True)
```

Comment:
This is a good point and I think this gets to your earlier comment
> Major conceptual comment: this is a really interesting result! I think it might be good to make sure somehow that the difference in generic/other LV distributions isn't something specific to the difference between eADAGE and PLIER (i.e. that it's a meaningful difference reflected in the data/underlying biology). I think this is hard to confirm, but one idea is to try playing around with the thresholds (e.g. the 2.5 standard deviations) and show that the plots are roughly the same even with other thresholds. And even if they're different, that could be interesting too! Other than that I don't have any great ideas for how to get at this, and obviously if Casey and your collaborators are satisfied with it then feel free to ignore me.

The 2.5 cutoff was selected based on the publication. This cutoff resulted in each LV having 100-200 high weight genes. In order to make sure that the total number of high weight genes wasn't effecting the trend observed in the box plots I used a quantile cutoff for PLIER that returned roughly a similar number of high weight genes per LV. I also played around with this cutoff to see if the trend in the boxplot changed and the box plot was consistent. Adding more total high weight genes didn't effect the the relationship between generic vs other genes. But this effected the LV genes that had a high proportion of generic genes because this was the denominator used.

I definitely this you raise a very good point about wanting to make sure that the difference between the generic vs other LV distributions are due to some biology vs something specific about the models. I'll definitely have to think more about this result and make sure I'm not over-interpreting it. Another thing I was thinking to check is the coverage of enriched pathways in the LVs. I think if more PLIER LVs were enriched in pathways (as the LVs were constrained to) this would also make sense why high weight genes were found to contribute to fewer LVs (since the LVs have some specific function). Whereas if LVs were not significantly enriched in a pathway but have some biological meaning, high weight generic genes contributing to more LVs could make sense.

Reply:
> Another thing I was thinking to check is the coverage of enriched pathways in the LVs.

This is a good idea! I do think PLIER will likely have more pathway enrichment since it's designed to find pathway-enriched LVs (it's part of the loss function for PLIER and not for ADAGE, as far as I know), but it would be interesting to see the results.